tests/Unit/PaymentGateways/EventHandlers/DonationCancelledTest.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Give\Tests\Unit\PaymentGateways\EventHandlers;
+
+use Exception;
+use Give\Donations\Models\Donation;
+use Give\Donations\ValueObjects\DonationStatus;
+use Give\PaymentGateways\EventHandlers\DonationCancelled;
+use Give\Tests\TestCase;
+use Give\Tests\TestTraits\RefreshDatabase;
+
+/**
+ * @unreleased
+ */
+class DonationCancelledTest extends TestCase
+{
+    use RefreshDatabase;
+    
+    /**
+     * @unreleased
+     *
+     * @throws Exception
+     */
+    public function testShouldSetStatusToCancelled()
```

Comment:
I love that you added unit tests! Let's keep this going and make these complete by asserting the opposite of what is expected like `testShouldNotSetStatusToCancelled` or more specific like `testShouldNotSetStatusToCancelledWhenRecurring`.  

Since each event handler has additional logic to prevent the status from being updated - we want to add tests to make sure those scenarios are working properly 😄

Reply:
Resolved in 45a24096bac0d5515d8ffdbc7bfa80de2ba5ebe2